wherehows-web/app/components/last-saved-by.ts

```diff
@@ -0,0 +1,57 @@
+import Component from '@ember/component';
+import { classNames } from '@ember-decorators/component';
+import { computed } from '@ember-decorators/object';
+import moment from 'moment';
+
+@classNames('last-saved-by')
+export default class LastSavedBy extends Component {
+  /**
+   * The entity responsible for the time transaction
+   * @type {string}
+   * @memberof LastSavedBy
+   */
+  actor: string;
+
+  /**
+   * The time at which the related transaction occurred
+   * @type {(number | undefined)}
+   * @memberof LastSavedBy
+   */
+  time: number | undefined;
+
+  /**
+   * Maximum length of characters to display of the actor
+   * @type {number}
+   * @memberof LastSavedBy
+   */
+  maxCharLength: number;
+
+  /**
+   * Alternate title for the component
+   * @type {string}
+   * @memberof LastSavedBy
+   */
+  title: string;
+
+  constructor() {
+    super(...arguments);
+
+    // defaults
+    (this.actor && typeof this.actor === 'string') || (this.actor = 'unknown');
```

Comment:
Should we create a decorator for this?

Reply:
There is already a decorator [@argument](https://github.com/ember-decorators/argument) but it isn't compatible ([issue here](https://github.com/ember-decorators/argument/issues/65)) with TypeScript at the moment
However, once this emberjs [RFC](https://github.com/emberjs/rfcs/blob/master/text/0337-native-class-constructor-update.md) is implemented, it won't be needed